dask/array/core.py

```diff
@@ -924,9 +924,10 @@ def store(sources, targets, lock=True, regions=None, compute=True,
     targets_dsk = Delayed.__dask_optimize__(targets_dsk, targets_keys)
 
     load_stored = (return_stored and not compute)
+    toks = [str(uuid.uuid1()) for _ in range(len(sources))]
     store_dsk = sharedict.merge(*[
-        insert_to_ooc(s, t, lock, r, return_stored, load_stored)
-        for s, t, r in zip(sources2, targets2, regions)
+        insert_to_ooc(s, t, lock, r, return_stored, load_stored, tok)
+        for s, t, r, tok in zip(sources2, targets2, regions, toks)
```

Comment:
Would it be possible for us to pick up the names used to store these objects instead of hanging onto the UUIDs used to generate them? Admittedly the syntax I've added here makes that a bit hard. Feel free to change this or the `insert_to_ooc` function as needed.

Reply:
I don't think so, not easily - the input array name is not useful, because we want to store to multiple outputs, and the outputs don't have names and at best would be hashed. I think uuid is good, and ensures that the write happens even when repeating the identical operation - that seems like the right thing to do to me for something that is essentially a side-effect operation.